conan/tools/system/package_manager.py

```diff
@@ -0,0 +1,208 @@
+import platform
+
+from conans.client.graph.graph import CONTEXT_BUILD
+from conans.errors import ConanException
+
+
+class SystemPackageManagerTool(object):
+    install_methods = ["install", "update"]
+    mode_check = "check"
+    tool_name = None
+    install_command = ""
+    update_command = ""
+    check_command = ""
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._active_tool = self._conanfile.conf["tools.system.package_manager:tool"] or self.get_default_tool()
+        self._sudo = self._conanfile.conf["tools.system.package_manager:sudo"]
+        self._sudo_askpass = self._conanfile.conf["tools.system.package_manager:sudo_askpass"]
+        self._mode = self._conanfile.conf["tools.system.package_manager:mode"] or self.mode_check
+        self._arch = self._conanfile.settings_build.get_safe('arch') \
+            if self._conanfile.context == CONTEXT_BUILD else self._conanfile.settings.get_safe('arch')
+        self._arch_names = {}
+        self._arch_separator = ""
+
+    def get_default_tool(self):
+        os_name = platform.system()
```

Comment:
It is possible that at some point we will want to check that it matches ``self.settings`` too, but it is ok as-is now.